examples/spring-boot-2.3/src/test/kotlin/com/linecorp/kotlinjdsl/spring/data/example/BookControllerIntegrationTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.linecorp.kotlinjdsl.spring.data.example
+
+import com.linecorp.kotlinjdsl.spring.data.example.entity.Book
+import org.assertj.core.api.WithAssertions
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.get
+import org.springframework.test.web.servlet.post
+
+@SpringBootTest
+@AutoConfigureMockMvc
+internal class BookControllerIntegrationTest : WithAssertions {
+    @Autowired
+    private lateinit var mockMvc: MockMvc
+
+    private val context = "/api/v1/books"
+
+    @Test
+    fun createBook() {
+        createBookTest(BookService.CreateBookSpec("name"))
+    }
+
+    @Test
+    fun findById() {
+        val spec = BookService.CreateBookSpec("name1")
+        val id = createBookTest(spec)
+        mockMvc.get("$context/$id")
+            .andExpect {
+                status { isOk }
+                content {
+                    json(Book(id = id, name = spec.name).toJson())
+                }
+            }
+    }
+
+    @Test
+    fun findByName() {
+        val spec1 = BookService.CreateBookSpec("name2")
+        val spec2 = BookService.CreateBookSpec("name2")
+        val id1 = createBookTest(spec1)
+        val id2 = createBookTest(spec2)
+        mockMvc.get(context) {
+            param("name", spec1.name)
+        }
+            .andExpect {
+                status { isOk }
+                content {
+                    json(
+                        buildString {
+                            append("[")
+                            append(Book(id = id1, name = spec1.name).toJson())
+                            append(",")
+                            append(Book(id = id2, name = spec2.name).toJson())
+                            append("]")
+                        }
+                    )
+                }
+            }
+    }
+
+    private fun createBookTest(spec: BookService.CreateBookSpec) = mockMvc.post(context) {
+        contentType = MediaType.APPLICATION_JSON
+        content = spec.toJson()
+    }.andExpect { status { isOk } }
+        .andReturn().response.contentAsString.toLong()
```

Comment:
-- korean
verify 절을 파라미터로 받아서 verify는 createBook 메소드에서 하면 좋을 것 같습니다.

-- english
It would be good to receive the verify clause as a parameter and verify it in the `createBook` method.

```kotlin
val id = createBook(BookService.CreateBookSpec("name")) {
  .andExpect { status { isOk } }
}

assertThat(id).isNotNull()
```

Reply:
-- korean
해당 메소드는 다른 메소드에서 다양하게 사용되고 있어서 verify 를 매번 받아서 하는건 옳은 것 같지 않습니다.

-- english
This method is used in various ways in other methods, so it doesn't seem right to receive verify every time.

Reply:
-- korean
네, 다른 메소드에서 다양하게 사용되고 있기 때문에 verify를 한 곳에만 집중시키는 것이 좋을 것 같습니다.
verify 절에 default 값으로 {} 를 주면 다른 메소드에서는 createBook() 을 통해 생성만 하고 verify는 하지 않게 된다고 생각했습니다.

-- english
Since it is used in various ways in different methods, it would be better to focus verification in one place.
If you give {} as a default value in the verify clause, other methods will only create it through createBook() and not verify.

Reply:
-- korean
제가 이야기 하는 것은 verify 를 할 수 있는 가능성을 메소드 안에서 준다면
다른 재사용 하는곳에서도 사용하지 않는다는 것은 코드의 흐름상 이상하게 보일 수 있다는 것 입니다. verify 를 하지 않고 넘어가고 그 결과로 그냥 다음 결과를 확인한다는 것이 어색합니다.
-- english
What I'm talking about is if you give the possibility of verifying inside the method
The fact that it is not used in other reuse places is that it can look strange in the flow of code. It's awkward to skip the verify and just check the next result as a result.

Reply:
-- korean
하나의 테스트에서 2가지 내용을 검증할 필요는 없다고 생각합니다.
createBook 테스트가 통과한다면 findById 테스트에서 사용하는 createBook 메소드는 이상이 없다는 증거가 됩니다.

그렇기 때문에 findById에서 createBookt 메소드가 정상적으로 동작한다는 것을 검증할 필요는 없을 것이라고 생각합니다.

-- english
There is no necessary to verify two logic in one test.
If the `createBook` test passes, the `createBook` method used in the `findById` test is evidence that there is no abnormality.

Therefore, I don't think it's necessary to verify that the `createBook` method works normally in findById.

Reply:
-- korean
그리고 메소드 명이 createBookTest 이기 때문에 BookTest를 만드는 것인지 createBoot 을 테스트 한다는 것인지 전달이 안 되는 점도 있습니다.

-- english
And because the method name is `Create Book Test`, it is hard to tell whether it is `creating Book Test` or `testing Create Boot`.

Reply:
-- korean
그 생각도 일리가 있는 것 같습니다. 반영했습니다.

-- english
That thought seems to make sense. applied.

Reply:
-- korean
createBookTest 메소드명도 createBook으로 변경하면 좋을 것 같습니다.

-- english
It would be good to change the `createBookTest` method name to `createBook`.

Reply:
-- korean
반영했습니다. assertThat을 넣지 않은 이유는 contentAsString.toLong() 으로 이미 문제가 있을 경우 예외가 발생하여 넣지 않았으니 참고해주세요.

-- english
applied. The reason for not putting assertThat is because if there is already a problem with contentAsString.toLong() , an exception is raised, so please refer to it.